packages/@lwc/engine-core/src/framework/component.ts

```diff
@@ -92,6 +96,9 @@ export function renderComponent(vm: VM): VNodes {
     }
 
     vm.tro.reset();
+    if (lwcRuntimeFlags.ENABLE_EXPERIMENTAL_SIGNALS) {
+        unsubscribeFromSignals(vm.component);
```

Comment:
Call unsubscribe wherever we reset the template reactive observer.

Reply:
I would make this a comment in the code.

Reply:
Yes. I would also call out the reason. E.g. this scenario:

1. the signal is only consumed inside a `lwc:if`
2. the `lwc:if` condition is initially true (and thus the component subscribes to the signal)
3. the `lwc:if` condition later changes to false (the signal is no longer consumed by the template, but we are still subscribed to updates)
5. the signal's value updates, causing the component to re-render (even though we won't use the new value)